### ISSUE TITLE
Access of mocked files should be normalized before check (/foo///bar)

### ISSUE THREAD

**Issue by atoomic** [cpanel/Test-MockFile#90]
When mocking `/file` then `//file` claim to be unmocked

we would expect all these tests to pass, all the most it s not possible to mock `//file` as this lead to  `Repeated forward slashes in path` error 

```perl
#!perl

use Test::More;
use Test::MockFile qw{strict};

my $f = q[/some/where];
my $mocked = Test::MockFile->file( $f => q[..] );

ok -f $f;
ok -f "/$f";
ok -f "////$f";
```

```
╰─> perl test.pl
ok 1
Use of stat to access unmocked file or directory '//some/where' in strict mode at test.pl line 10 at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/Test/MockFile.pm line 238.
	Test::MockFile::_strict_mode_violation("stat", ARRAY(0x1ff89c8)) called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/Test/MockFile.pm line 1136
	Test::MockFile::_real_file_access_hook("stat", ARRAY(0x1ff89c8)) called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/Test/MockFile.pm line 610
	Test::MockFile::_mock_stat("stat", "//some/where") called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/x86_64-linux-64int/Overload/FileCheck.pm line 295
	Overload::FileCheck::_check_from_stat("f", "//some/where", CODE(0x1ec28c0)) called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/x86_64-linux-64int/Overload/FileCheck.pm line 265
	Overload::FileCheck::__ANON__("f", "//some/where") called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/x86_64-linux-64int/Overload/FileCheck.pm line 210
	Overload::FileCheck::__ANON__("//some/where") called at /usr/local/cpanel/3rdparty/perl/532/lib/perl5/cpanel_lib/x86_64-linux-64int/Overload/FileCheck.pm line 586
	Overload::FileCheck::_check(286, "//some/where") called at test.pl line 10
# Tests were run but no plan was declared and done_testing() was not seen.
# Looks like your test exited with 255 just after 1.
```

**Comment by atoomic** [comment]
same issue when mocking a directory `/a/b/c` we should expect to use `/a////b//c////`

**Comment by xsawyerx** [comment]
Would it be correct to rephrase this as "We should normalize all paths to remove multiple forward slashes"?

**Comment by atoomic** [comment]
I agree T::MF should normalize all paths

**Comment by xsawyerx** [comment]
I don't think it should normalize all paths. I'm just wondering if this is a good representation of this ticket.

**Comment by toddr** [comment]
We have a lot of code that is passing file paths with /foo//bar. These double slashes need to be stripped when passed to open, glob,  and opendir, etc.

**Comment by atoomic** [comment]
for example to test this code

```perl
sub to_test($a="here", $b="", $c="there") {
   my $path = "/$a/$b/$c";
   ... # do something with path
}
```

We could not test the function using T::MF without updating the function itself ( which is probably pointing to a bug )
